lib/commands/command-plugin-update.bash

```diff
@@ -2,23 +2,35 @@
 
 plugin_update_command() {
   if [ "$#" -lt 1 ]; then
-    display_error "usage: asdf plugin-update {<name> | --all} [git-ref]"
+    display_error "usage: asdf plugin-update {<name> [git-ref] | --all}"
     exit 1
   fi
 
   local plugin_name="$1"
-  local gitref="${2:-master}"
+  local plugin_path
+  local branch_of_head
+  local gitref
+
   if [ "$plugin_name" = "--all" ]; then
     for dir in "$(asdf_data_dir)"/plugins/*; do
-      echo "Updating $(basename "$dir")..."
-      (cd "$dir" && git fetch -p -u origin "$gitref:$gitref" && git checkout -f "$gitref")
+      branch_of_head=$(git --git-dir "$dir/.git" rev-parse --abbrev-ref HEAD 2>/dev/null)
+
+      printf "%s %s%s\\n" "Updating" "$(basename "$dir")" "..."
+      gitref="${branch_of_head}"
+      git --git-dir "$dir/.git" fetch -p -u origin "$gitref:$gitref"
+      git --git-dir "$dir/.git" checkout -f "$gitref"
+      # (cd "$dir" && git fetch -p -u origin "$gitref:$gitref" && git checkout -f "$gitref")
     done
   else
-    local plugin_path
-    plugin_path="$(get_plugin_path "$plugin_name")"
     check_if_plugin_exists "$plugin_name"
-    echo "Updating $plugin_name..."
-    (cd "$plugin_path" && git fetch -p -u origin "$gitref:$gitref" && git checkout -f "$gitref")
+    plugin_path="$(get_plugin_path "$plugin_name")"
+    branch_of_head=$(git --git-dir "$plugin_path/.git" rev-parse --abbrev-ref HEAD 2>/dev/null)
+
+    printf "%s %s%s\\n" "Updating" "$plugin_name" "..."
+    gitref="${2:-${branch_of_head}}"
```

Comment:
I haven't tested if this syntax works as I suspect. I can never remember.

Reply:
Yes, that's the correct syntax to use the second parameter of the `plugin_update_command` function and fall back to the value of `$branch_of_head` if it's unset or empty. 😉